tests/runner.py

```diff
@@ -252,16 +260,10 @@ def build(self, src, dirname, filename, output_processor=None, main_file=None, a
       output_processor(open(filename + '.o.js').read())
 
     if self.emcc_args is not None:
-      if '--memory-init-file' in self.emcc_args:
-        memory_init_file = int(self.emcc_args[self.emcc_args.index('--memory-init-file')+1])
-      else:
-        memory_init_file = ('-O2' in self.emcc_args or '-O3' in self.emcc_args or '-Oz' in self.emcc_args) and not Settings.SIDE_MODULE
       src = open(filename + '.o.js').read()
-      if memory_init_file:
+      if self.uses_memory_init_file():
         # side memory init file, or an empty one in the js
         assert ('/* memory initializer */' not in src) or ('/* memory initializer */ allocate([]' in src)
-      else:
-        assert 'memory initializer */' in src or '/*' not in src # memory initializer comment, or cleaned-up source with no comments
```

Comment:
@kripken: Judging from your lack of comment on this line, I take it that dropping this particular check is OK?